From fbe7a335332c6afc676174a47bf109e2c8cf1647 Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Wed, 4 Mar 2026 14:47:01 -0500
Subject: [PATCH 09/13] Issue 7271 - Add new plugin pre-close function check to
 plugin_invoke_plugin_pb

Description:

In plugin_invoke_plugin_pb we were not checking for the new pre-close function
which led to an error in the logs: pb_op is NULL. In a debug build this leads
to an assertion error at shutdown.

relates: https://github.com/389ds/389-ds-base/issues/7271

Reviewed by: vashirov(Thanks!)
---
 ldap/servers/slapd/plugin.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ldap/servers/slapd/plugin.c b/ldap/servers/slapd/plugin.c
index 8b62dba32..ab7875e87 100644
--- a/ldap/servers/slapd/plugin.c
+++ b/ldap/servers/slapd/plugin.c
@@ -3674,6 +3674,7 @@ plugin_invoke_plugin_pb(struct slapdplugin *plugin, int operation, Slapi_PBlock
     if (operation == SLAPI_PLUGIN_START_FN ||
         operation == SLAPI_PLUGIN_POSTSTART_FN ||
         operation == SLAPI_PLUGIN_CLOSE_FN ||
+        operation == SLAPI_PLUGIN_PRE_CLOSE_FN ||
         operation == SLAPI_PLUGIN_CLEANUP_FN ||
         operation == SLAPI_PLUGIN_BE_PRE_CLOSE_FN ||
         operation == SLAPI_PLUGIN_BE_POST_OPEN_FN ||
-- 
2.52.0

